share/eds/byzantine/bad_encoding.go

```diff
@@ -167,31 +185,45 @@ func (p *BadEncodingProof) Validate(hdr libhead.Header) error {
 		shares[index] = share.GetData(shr.Share)
 	}
 
-	odsWidth := uint64(len(merkleRoots) / 2)
 	codec := share.DefaultRSMT2DCodec()
 
-	// rebuild a row or col.
+	// We assume that the proof is valid in case we proved the inclusion of `Shares` but
+	// the row/col can't be reconstructed, or the building of NMTree fails.
 	rebuiltShares, err := codec.Decode(shares)
 	if err != nil {
-		return err
+		log.Infow("failed to decode shares at height",
```

Comment:
Non-blocking, can improve in a follow-up issue. Can this and the below info logs clarify that the failure is a `good thing`? Otherwise a user might think something went wrong.

Reply:
That's why we make an info log level(and not an error). Or do you think it's not enough?

Reply:
You're correct that fail with info instead of warn or error _should_ be sufficient, but IMO it's not obvious enough to users. Making it explicit that this failure implies the fraud proof is valid is clearer. But as above, you don't need to fix this in this PR if you're not convinced, you can file a `good first issue`.

Reply:
Thanks for the suggestion. I will make an issue